src/core/server/saved_objects/migrationsv2/integration_tests/type_registrations.test.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { REMOVED_TYPES } from '../../migrations/core';
+import * as kbnTestServer from '../../../../test_helpers/kbn_server';
+
+// Types should NEVER be removed from this array
+const previouslyRegisteredTypes = [
```

Comment:
I ran this test on all branches from 7.7 forward. We can and should do it for the prior 7.x versions but it will take some extra work since the typeRegistry API didn't exist yet. Will do as a follow up.